cmd/commands/reset/command.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package reset
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/mysteriumnetwork/node/config"
+	"github.com/mysteriumnetwork/node/core/auth"
+	"github.com/mysteriumnetwork/node/core/node"
+	"github.com/mysteriumnetwork/node/core/storage/boltdb"
+	"github.com/urfave/cli/v2"
+)
+
+// flagResetTequilapiAuth instructs to reset Tequilapi auth.
+var flagResetTequilapiAuth = cli.BoolFlag{
+	Name: "tequilapi",
+	Usage: fmt.Sprintf("Reset Tequilapi auth credentials to values provider in --%s and --%s flags",
+		config.FlagTequilapiUsername.Name,
+		config.FlagTequilapiPassword.Name,
+	),
+	Value: true,
+}
+
+// NewCommand creates reset command.
+func NewCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "reset",
+		Usage:     "Resets Mysterium Node to defaults",
+		ArgsUsage: " ",
+		Flags:     []cli.Flag{&flagResetTequilapiAuth},
+		Action: func(ctx *cli.Context) error {
+			config.ParseFlagsNode(ctx)
+			nodeOptions := node.GetOptions()
+
+			cmd, err := newResetCommand(ctx.App.Writer, nodeOptions.Directories)
+			if err != nil {
+				return err
+			}
+
+			return cmd.Run(ctx)
+		},
+	}
+}
+
+// newResetCommand creates instance of reset command.
+func newResetCommand(
+	writer io.Writer,
+	dirOptions node.OptionsDirectory,
+) (*resetCommand, error) {
+	fmt.Println(dirOptions.Storage)
+
+	if err := dirOptions.Check(); err != nil {
+		return nil, err
+	}
+
+	localStorage, err := boltdb.NewStorage(dirOptions.Storage)
```

Comment:
Am I wrong or is this never closed? Can that introduce issues for us?

Reply:
Improved